### PR TITLE
[SM-574] Hide edit/delete when user only has read access

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [x] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Hide write actions when user only has read access.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: bitwarden_license/bit-web/src/app/secrets-manager/secrets/dialog/secret-dialog.component.ts [hunk 1]
```diff
@@ -80,6 +81,11 @@ export class SecretDialogComponent implements OnInit {
       notes: secret.note,
       project: secret.projects[0]?.id ?? "",
     });
+
+    if (!secret.write) {
+      this.formGroup.disable();
```
Comment (willmartian) [inline bitwarden_license/bit-web/src/app/secrets-manager/secrets/dialog/secret-dialog.component.ts:86]: I would pull in the latest changes from master to ensure that the form disable/enable logic from https://github.com/bitwarden/clients/pull/4958 doesn't conflict with this.

File: bitwarden_license/bit-web/src/app/secrets-manager/secrets/dialog/secret-dialog.component.html [hunk 2]
```diff
@@ -38,7 +38,14 @@
...
         {{ "save" | i18n }}
       </button>
-      <button type="button" bitButton buttonType="secondary" bitFormButton bitDialogClose>
+      <button
+        type="button"
+        bitButton
+        buttonType="secondary"
+        bitFormButton
+        bitDialogClose
+        [disabled]="false"
```
Comment (willmartian) [inline bitwarden_license/bit-web/src/app/secrets-manager/secrets/dialog/secret-dialog.component.html:47]: Is the purpose of this and the changes in `libs/components/src/async-actions/form-button.directive.ts` just to keep this button from getting disabled when the form group is disabled? 

Can't we also just remove the `bitFormButton` directive to achieve the same thing?

Comment (willmartian) [reply bitwarden_license/bit-web/src/app/secrets-manager/secrets/dialog/secret-dialog.component.html:47]: Makes sense 👍

**Review by Thomas-Avery — COMMENTED** [review]
Looks good.

Should we hide the bulk delete button if the user has no write access to any secret in the list? 


bitwarden_license/bit-web/src/app/secrets-manager/shared/secrets-list.component.html
```
</button>
  <button type="button" bitMenuItem (click)="bulkDeleteSecrets()">
    <i class="bwi bwi-fw bwi-trash tw-text-danger" aria-hidden="true"></i>
    <span class="tw-text-danger">{{ "deleteSecrets" | i18n }}</span>
  </button>
```

**Review by Thomas-Avery — APPROVED** [review]
LGTM

**Pull request merged**